rust/arrow/src/csv/reader.rs

```diff
@@ -77,15 +70,20 @@ fn infer_field_schema(string: &str) -> DataType {
         return DataType::Utf8;
     }
     // match regex in a particular order
-    if BOOLEAN_RE.is_match(string) {
-        DataType::Boolean
-    } else if DECIMAL_RE.is_match(string) {
-        DataType::Float64
-    } else if INTEGER_RE.is_match(string) {
-        DataType::Int64
-    } else {
-        DataType::Utf8
+    let lower = string.to_ascii_lowercase();
+    if lower == "true" || lower == "false" {
+        return DataType::Boolean;
+    }
+    let skip_minus = if string.starts_with('-') { 1 } else { 0 };
```

Comment:
One thing I I wonder is does this code handle invalid data like `12.12.12` (aka if `split` returns more than 1 decimal)?

But given that this is just code for inferring schema, it is probably ok if that gets identified incorrect as a float...

Reply:
Yeah... check could be added if it's too inclusive, I think easily using https://doc.rust-lang.org/std/primitive.str.html#method.splitn

Reply:
Added